src/utils/dataStructureUtils.ts

```diff
@@ -0,0 +1,28 @@
+function createMapFromObject(obj: Record<string, any>) {
+  const map = new Map<string, any>();
+
+  for (const key in obj) {
+    map.set(key, obj[key]);
+  }
+
+  return map;
+}
+
+function isArrayOfString(x: unknown): boolean {
+  return (
+    Array.isArray(x) &&
+    x.every(item => Boolean(item && typeof item === "string"))
```

Comment:
Good catch, no. It should return true if `x = ["", "", "foo"]`. Otherwise, we cannot handle the error 
```suggestion
    x.every(item => typeof item === "string")
```